src/models/AbstractWidget.php

```diff
@@ -0,0 +1,126 @@
+<?php namespace Leean\Widgets\Models;
+
+use Leean\Acf\All;
+
+/**
+ * Class AbstractWidget.
+ *
+ * @package Leean\Widgets\Models
+ */
+class AbstractWidget extends \WP_Widget
+{
+	/**
+	 * The slug.
+	 *
+	 * @var bool|string
+	 */
+	private $_slug = '';
+
+	/**
+	 * AbstractWidget constructor.
+	 *
+	 * @param string $title       Widget title
+	 * @param string $description Widget description.
+	 * @param bool   $slug		  Widget slug (use class name if not supplied).
+	 */
+	public function __construct( $title, $description, $slug = false ) {
```

Comment:
Yes since on java you can create an abstract class from another abstract class. 

Well I would say if the class aim is just share behaviors we should use plain inheritance.